examples/hybrid/TurbulenceConvectionUtils.jl

```diff
@@ -102,31 +101,81 @@ function init_tc!(Y, p, params, colidx)
-    C123 = CCG.Covariant123Vector
-    t = FT(0)
 
-    # TODO: can we simply remove this?
-    If = CCO.InterpolateC2F(bottom = CCO.Extrapolate(), top = CCO.Extrapolate())
-    @. p.edmf_cache.aux.face.ρ[colidx] = If(Y.c.ρ[colidx])
+    # parent(state.aux.face) .= NaN
+    # parent(state.aux.cent) .= NaN
 
-    # TODO: convert initialize_profiles to set prognostic state, not aux state
+    # Initialization of auxiliary variables:
+    (; ᶜp, ᶜts) = p
+    ᶜθ_liq_ice = p.edmf_cache.aux.cent.θ_liq_ice
+    ᶜq_tot = p.edmf_cache.aux.cent.q_tot # only initialized for moist profiles
+    ᶜtke = p.edmf_cache.aux.cent.tke
+    ᶜHvar = p.edmf_cache.aux.cent.Hvar # only initialized for GABLS
+    ᶜHvar .= 0
     Cases.initialize_profiles(case, grid, thermo_params, state)
+    if p.atmos.moisture_model isa CA.DryModel
+        @. ᶜts[colidx] =
+            TD.PhaseDry_pθ(thermo_params, ᶜp[colidx], ᶜθ_liq_ice[colidx])
+    else
+        @. ᶜts[colidx] = TD.PhaseEquil_pθq(
+            thermo_params,
+            ᶜp[colidx],
+            ᶜθ_liq_ice[colidx],
+            ᶜq_tot[colidx],
+        )
+    end
 
-    # Temporarily, we'll re-populate ρq_tot based on initial aux q_tot
-    q_tot = p.edmf_cache.aux.cent.q_tot[colidx]
-    @. Y.c.ρq_tot[colidx] = Y.c.ρ[colidx] * q_tot
-    set_thermo_state_pθq!(Y, p, colidx)
-    set_grid_mean_from_thermo_state!(thermo_params, state, grid)
-    assign_thermo_aux!(state, grid, edmf.moisture_model, thermo_params)
-    initialize_edmf(edmf, grid, state, surf_params, tc_params, t)
+    # Initialization of prognostic variables:
+    (; ᶜinterp) = p.operators
+    C123 = CCG.Covariant123Vector
+    @. Y.c.ρ[colidx] = TD.air_density(thermo_params, ᶜts[colidx])
+    @. Y.c.ρe_tot[colidx] =
+        Y.c.ρ[colidx] * TD.total_energy(
+            thermo_params,
+            ᶜts[colidx],
+            LA.norm_sqr(C123(Y.c.uₕ[colidx]) + C123(ᶜinterp(Y.f.w[colidx]))) /
+            2,
+            p.ᶜΦ[colidx],
+        )
+    if !(p.atmos.moisture_model isa CA.DryModel)
+        @. Y.c.ρq_tot[colidx] = Y.c.ρ[colidx] * ᶜq_tot[colidx]
+    end
+
+    N_up = TC.n_updrafts(edmf)
+    a_min = edmf.minimum_area
+    lg = CC.Fields.local_geometry_field(axes(Y.f))
+    @inbounds for i in 1:N_up
+        @. Y.c.turbconv.up[i].ρarea[colidx] = Y.c.ρ[colidx] * a_min
+        @. Y.c.turbconv.up[i].ρaq_tot[colidx] =
+            Y.c.turbconv.up[i].ρarea[colidx] * ᶜq_tot[colidx]
+        @. Y.c.turbconv.up[i].ρaθ_liq_ice[colidx] =
+            Y.c.turbconv.up[i].ρarea[colidx] * ᶜθ_liq_ice[colidx]
+        @. Y.f.turbconv.up[i].w[colidx] =
+            CC.Geometry.Covariant3Vector(CC.Geometry.WVector(FT(0)), lg[colidx])
+    end
+    a_en = (1 - N_up * a_min)
+    @. Y.c.turbconv.en.ρatke[colidx] = Y.c.ρ[colidx] * a_en * ᶜtke[colidx]
+    if p.atmos.turbconv_model.thermo_covariance_model isa
+        TC.PrognosticThermoCovariances
+        @. Y.c.turbconv.en.ρaHvar[colidx] = Y.c.ρ[colidx] * a_en * ᶜHvar[colidx]
```

Comment:
Would you also need the `QTvar` and `HQTcov`?

That's what is deleted from initial conditions below:
```
    if edmf.thermo_covariance_model isa TC.PrognosticThermoCovariances
        prog_en.ρaHvar .= aux_gm.Hvar .* ρ_c .* ae
        prog_en.ρaQTvar .= aux_gm.QTvar .* ρ_c .* ae
        prog_en.ρaHQTcov .= aux_gm.HQTcov .* ρ_c .* ae
```

Reply:
Yes, but these other values weren't being initialized in `initialize_profiles`. I've added them in anyway, but set them to 0.